code/_onclick/hud/action.dm

```diff
@@ -135,12 +135,12 @@
 	var/image/img
 	if(owner.action_type == AB_ITEM && owner.target)
 		var/obj/item/I = owner.target
-		img = image(I.icon, src , I.icon_state)
+		overlays += I.appearance
```

Comment:
@phil235 , @Jordie0608 is using a 508 feature, the ability to grab internal appearances (cached collections of underlays, overlays, icon, icon_state, and other visual data) in one variable, and swap them/add them to other objects.